docs/design/fabric-v2+/attestation-interface.md

```diff
@@ -0,0 +1,63 @@
+# Attestation Interface
+
+This document defines the interface for producing and verifying an attestation.
+
+
+## Producing an attestation
+
+This interface is only available inside an enclave.
+
+```
+int init_attestation(
+    uint8_t* params,
+    uint32_t params_length);
+
+```
+```
+int get_attestation(
+    uint8_t statement[32],
```

Comment:
I guess you wanted to keep the API not FPC specific so it could more widely reused (e.g., in PDO)? This makes sense.  I wonder, though, whether you might not make it an arbitary-length byte array and then hash it internally to 32 bytes). Otherwise, you force the caller in most cases (e.g., fpc) to also have to do crypto on top.  I think it would be nicer if the caller only would have to do serialization?

Reply:
> I guess you wanted to keep the API not FPC specific so it could more widely reused (e.g., in PDO)?

Desirable, but that's secondary. The primary reason is that no FPC-specific thing should belong to a "common attestation" api, but rather they should be implemented above it.

> whether you might not make it an arbitary-length byte array and then hash it internally to 32 bytes?

First, I would argue that it is an implementation detail -- we could also make it optional.
Second, the rationale for not doing it is to avoid forcing a hash. The caller could choose to simply put an arbitrary 32-byte array.

> Otherwise, you force the caller in most cases (e.g., fpc) to also have to do crypto on top

Yes, that's the trade-off. Arguably a hash function is usually available.

Reply:
changed to arbitrary length statement